tools/devel-server.py

```diff
@@ -90,133 +96,131 @@ def serve_front(self):
 
 * [Available puzzles](/puzzles)
 * [Raw filesystem view](/files/)
-* [Documentation](/files/doc/)
-* [Instructions](/files/doc/devel-server.md) for using this server
+* [Documentation](/files/docs/)
+* [Instructions](/files/docs/devel-server.md) for using this server
 
 If you use this development server to run a contest,
 you are a fool.
 """
-        self.serve_md(page)
+        payload = mdpage(body).encode('utf-8')
+        self.send_response(HTTPStatus.OK)
+        self.send_header("Content-Type", "text/html; charset=utf-8")
+        self.send_header("Content-Length", len(payload))
+        self.end_headers()
+        self.wfile.write(payload)
 
     def serve_puzzles(self):
-        body = []
+        body = io.StringIO()
         path = self.path.rstrip('/')
         parts = path.split("/")
-        #raise ValueError(parts)
-        if len(parts) < 3:
-            # List all categories
-            body.append("# Puzzle Categories")
-            for i in glob.glob(os.path.join("puzzles", "*", "")):
-                body.append("* [{}](/{})".format(i, i))
-            self.serve_md('\n'.join(body))
-            return
+        title = None
+        fpath = None
+        points = None
+        cat = None
+        puzzle = None
 
-        fpath = os.path.join("puzzles", parts[2])
-        cat = moth.Category(fpath, seed)
-        if len(parts) == 3:
+        try:
+            fpath = os.path.join("puzzles", parts[2])
+            points = int(parts[3])
+        except:
+            pass
+
+        if fpath:
+            cat = moth.Category(fpath, seed)
+        if points:
+            puzzle = cat.puzzle(int(parts[3]))
```

Comment:
Why not `puzzle = cat.puzzle(points)`?  Might be worthwhile to use cat and points later on to drive self documenting code.